OctoAwesome/OctoAwesome.Runtime/GameService.cs

```diff
@@ -0,0 +1,157 @@
+        /// <param name="cache"><see cref="ILocalChunkCache"/> as workspace</param>
+        /// <param name="radius">Radius der <see cref="Entity"/></param>
+        /// <param name="height">Höhe der <see cref="Entity"/></param>
+        /// <param name="deltaPosition">Positionsänderung zwischen zwei Simulationsdurchläufen</param>
+        /// <param name="velocity">Berechnete Geschwindigkeit</param>
+        /// <exception cref="ArgumentNullException">Cache</exception>
+        /// <returns>Geschwindigkeit der <see cref="Entity"/> nach der Killisionsprüfung</returns>
+        public Vector3 WorldCollision(GameTime gameTime, Coordinate position, ILocalChunkCache cache, float radius, float height, 
+            Vector3 deltaPosition, Vector3 velocity)
+        {
+            if (cache == null)
+                throw new ArgumentNullException(nameof(cache));
+
+            Vector3 move = deltaPosition;
+
+            //Blocks finden die eine Kollision verursachen könnten
+            int minx = (int) Math.Floor(Math.Min(
+                position.BlockPosition.X - radius,
+                position.BlockPosition.X - radius + deltaPosition.X));
+            int maxx = (int) Math.Ceiling(Math.Max(
+                position.BlockPosition.X + radius,
+                position.BlockPosition.X + radius + deltaPosition.X));
+            int miny = (int) Math.Floor(Math.Min(
+                position.BlockPosition.Y - radius,
+                position.BlockPosition.Y - radius + deltaPosition.Y));
+            int maxy = (int) Math.Ceiling(Math.Max(
+                position.BlockPosition.Y + radius,
+                position.BlockPosition.Y + radius + deltaPosition.Y));
+            int minz = (int) Math.Floor(Math.Min(
+                position.BlockPosition.Z,
+                position.BlockPosition.Z + deltaPosition.Z));
+            int maxz = (int) Math.Ceiling(Math.Max(
+                position.BlockPosition.Z + height,
+                position.BlockPosition.Z + height + deltaPosition.Z));
+
+            //Beteiligte Flächen des Spielers
+            var playerplanes = CollisionPlane.GetEntityCollisionPlanes(radius, height, velocity, position);
+
+            bool abort = false;
+
+            for (int z = minz; z <= maxz && !abort; z++)
+            {
+                for (int y = miny; y <= maxy && !abort; y++)
+                {
+                    for (int x = minx; x <= maxx && !abort; x++)
+                    {
+                        move = velocity * (float) gameTime.ElapsedGameTime.TotalSeconds;
+
+                        Index3 pos = new Index3(x, y, z);
+                        Index3 blockPos = pos + position.GlobalBlockIndex;
+                        ushort block = cache.GetBlock(blockPos);
+                        if (block == 0) continue;
+
+                        var blockplane = CollisionPlane.GetBlockCollisionPlanes(pos, velocity);
+
+                        var planes = from pp in playerplanes
```

Comment:
Ich glaube das wurde vom Lassi ausgeliehen, Pfeilchen wären trotzdem schöner

Reply:
das bleibt mal so :)
was stört dich daran ?

Reply:
Ich denke, dass in unserem Code style Richtlinien drinsteht, dass LINQ immer mit Methoden verwendet wird. @Gallimathias meinte aber, dass es nicht so schlimm sei, weil es ja von @CsharpLassi übernommen wurde (der diese Schreibweise liebt).

Reply:
um ehrlich zu sein... ich hab es versucht und bin gescheitert wenn es einer von euch hinbekommt dann bitte. kann jetzt nur sagen das wird nicht mehr so kompakt bleiben wie es da steht. diese Schreibweise kann ein bisschen mehr als linq wie es aussieht, aber beweist mir doch das Gegenteil :)

Reply:
Am ende sollte LinQ meiner Meinung nach ganz raus. Vor allem bei allem kontinuierlichen. Oder vlt sogar gleich komplett verbieten. Könnte mich evtl an eine alternative setzen, die rein auf Listen und Arrays arbeiten kann, wäre dann nat. Nur methodensyntax

Reply:
erstmal Manuel hat recht. ^^

Dann die Pfeilchen auflösung sehe nur kurz hingerotzt so aus:

````CSharp
var planes = playerplanes
                              .SelectMany(pp => blockplane.Select(bp =>
                              {
                                  if (!CollisionPlane.Intersect(bp, pp))
                                      return null;

                                  var distance = CollisionPlane.GetDistance(bp, pp);
                                  if (CollisionPlane.CheckDistance(distance, move))
                                      return new { BlockPlane = bp, PlayerPlane = pp, Distance = distance };
                                  else
                                      return null;
                              }
                                  )
                                  .Where(a => a != null)
                              );
````

Aber ich pfliche Julian bei das man hier eigentlich Linq ausenvorlassen sollte. Das ist auch kein Problem weil die liste nur aus bequemlichkeit erstellt wurde, wirklich brauchen tun wir sie an der stelle eigentlich nicht ^^.

Reply:
Ess könnte so klappen, müsste aber noch gecheckt werden:

````CSharp
 //Beteiligte Flächen des Spielers
            var playerplanes = CollisionPlane.GetEntityCollisionPlanes(radius, height, velocity, position);

            for (int z = minz; z <= maxz; z++)
            {
                for (int y = miny; y <= maxy; y++)
                {
                    for (int x = minx; x <= maxx; x++)
                    {
                        move = velocity * (float)gameTime.ElapsedGameTime.TotalSeconds;

                        Index3 pos = new Index3(x, y, z);
                        Index3 blockPos = pos + position.GlobalBlockIndex;
                        ushort block = cache.GetBlock(blockPos);
                        if (block == 0) continue;

                        var blockplanes = CollisionPlane.GetBlockCollisionPlanes(pos, velocity);
                        
                        foreach (var playerPlane in playerplanes)
                        {
                            foreach (var blockPlane in blockplanes)
                            {
                                if (!CollisionPlane.Intersect(blockPlane, playerPlane))
                                    continue;

                                var distance = CollisionPlane.GetDistance(blockPlane, playerPlane);
                                if (!CollisionPlane.CheckDistance(distance, move))
                                    continue;

                                var subvelocity = (distance / (float)gameTime.ElapsedGameTime.TotalSeconds);
                                var diff = velocity - subvelocity;

                                float vx;
                                float vy;
                                float vz;

                                if (blockPlane.normal.X != 0 && (velocity.X > 0 && diff.X >= 0 && subvelocity.X >= 0 ||
                                    velocity.X < 0 && diff.X <= 0 && subvelocity.X <= 0))
                                    vx = subvelocity.X;
                                else
                                    vx = velocity.X;

                                if (blockPlane.normal.Y != 0 && (velocity.Y > 0 && diff.Y >= 0 && subvelocity.Y >= 0 ||
                                    velocity.Y < 0 && diff.Y <= 0 && subvelocity.Y <= 0))
                                    vy = subvelocity.Y;
                                else
                                    vy = velocity.Y;

                                if (blockPlane.normal.Z != 0 && (velocity.Z > 0 && diff.Z >= 0 && subvelocity.Z >= 0 ||
                                    velocity.Z < 0 && diff.Z <= 0 && subvelocity.Z <= 0))
                                    vz = subvelocity.Z;
                                else
                                    vz = velocity.Z;

                                velocity = new Vector3(vx, vy, vz);
                                if (vx == 0 && vy == 0 && vz == 0)
                                {
                                    return velocity;
                                }
                            }
                        }
                    }
                }
            }
            return velocity;
````

Reply:
dreifach for mit zweifach foreach :) 
aber das letzte sollte laufen